package bacs.archive.pb.problem;

import "bacs/problem/problem.proto";

message IdSet
{
    repeated string ids = 1;
}

message InfoMap
{
    message Entry
    {
        required string id = 1;
        optional bacs.problem.Problem info = 2;
    }

    repeated Entry items = 1;
}

message HashMap
{
    message Entry
    {
        required string id = 1;
        optional bytes hash = 2;
    }

    repeated Entry items = 1;
}

message FlagSet
{
    repeated string flags = 1;
}

message Status
{
    repeated string flags = 1;
    required bytes hash = 2;
}

message StatusMap
{
    message Entry
    {
        required string id = 1;
        optional Status status = 2;
    }

    repeated Entry items = 1;
}

message ImportInfo
{
    optional string error = 1;
    optional Status status = 2;
}

message ImportMap
{
    message Entry
    {
        required string id = 1;
        required ImportInfo import_info = 2;
    }

    repeated Entry items = 1;
}
